wpt/tests/serialization/serialization.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2021 Google LLC.
+# Copyright (c) Microsoft Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+
+
+# Testing serialization.
+async def assert_serialization(js_str_object, expected_serialized_object,
+      send_blocking_command, context_id, recursive_compare):
+    result = await send_blocking_command("script.evaluate", {
+        "expression": f"({js_str_object})",
+        "target": {"context": context_id}})
+    # Compare ignoring `objectId`.
+    recursive_compare(expected_serialized_object, result, ["objectId"])
+
+
+@pytest.mark.asyncio
+async def test_serialization_undefined(send_blocking_command, context_id,
+      recursive_compare):
+    await assert_serialization("undefined", {
+        "type": "undefined"}, send_blocking_command, context_id,
+                               recursive_compare)
+
+
+@pytest.mark.asyncio
+async def test_serialization_null(send_blocking_command, context_id,
+      recursive_compare):
+    await assert_serialization("null", {
+        "type": "null"}, send_blocking_command, context_id, recursive_compare)
+
+
+@pytest.mark.asyncio
+async def test_serialization_string(send_blocking_command, context_id,
+      recursive_compare):
+    await assert_serialization("'someStr'", {
+        "type": "string",
+        "value": "someStr"}, send_blocking_command, context_id,
+                               recursive_compare)
+
+
+@pytest.mark.asyncio
+async def test_serialization_number(send_blocking_command, context_id,
+      recursive_compare):
+    await assert_serialization("123", {
```

Comment:
test more type of numbers like -123, -0.56, 1.2e10, -1.2e10, .123, 1.2e-10, -1.2e-10 ?

Reply:
I planned to add some tests here and to the WPT, so that provide a possibility to copy-paste tests to/from WPT repo. I'm not sure if the approach will be there or not. Full test coverage was out of scope for this PR.